src/VirtualClient/VirtualClient.Actions/SPECview/SpecViewExecutor.cs

```diff
@@ -0,0 +1,227 @@
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+    using VirtualClient.Contracts.Metadata;
+
+    /// <summary>
+    /// Executes the SPECView workload.
+    /// </summary>
+    [WindowsCompatible]
+    public class SpecViewExecutor : VirtualClientComponent
+    {
+        private const string VisualStudioCRuntimePackageName = "visualstudiocruntime";
+        private IFileSystem fileSystem;
+        private ISystemManagement systemManagement;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public SpecViewExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+            : base(dependencies, parameters)
+        {
+            this.fileSystem = dependencies.GetService<IFileSystem>();
+            this.systemManagement = dependencies.GetService<ISystemManagement>();
+        }
+
+        /// <summary>
+        /// The command line argument defined in the profile.
+        /// </summary>
+        public string CommandArguments
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(SpecViewExecutor.CommandArguments));
+            }
+        }
+
+        /// <summary>
+        /// The viewset that will be run by SPECviewperf.
+        /// </summary>
+        public string Viewset
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(SpecViewExecutor.Viewset));
+            }
+        }
+
+        /// <summary>
+        /// The path to the RunViewperf.exe.
+        /// </summary>
+        public string ExecutablePath { get; set; }
+
+        /// <summary>
+        /// Defines the path to the SPECview package that contains the workload
+        /// executable.
+        /// </summary>
+        protected DependencyPath Package { get; set; }
+
+        /// <summary>
+        /// Initializes the environment
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            PlatformSpecifics.ThrowIfNotSupported(this.Platform);
+
+            await this.InitializePackageLocationAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            this.ExecutablePath = this.PlatformSpecifics.Combine(this.Package.Path, "RunViewperf.exe");
+        }
+
+        /// <summary>
+        /// Executes the SPECview workload.
+        /// </summary>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            IList<Metric> metrics = new List<Metric>();
+
+            string viewset = this.GenerateCommandArguments(this.Viewset);
```

Comment:
You use the viewset to generate your arguments but also have this.CommandArguments, this is a little confusing, could you check again?